docs/pages/guides/hello-world/filter-sync.mdx

```diff
@@ -0,0 +1,224 @@
+        key0: "0x" + "5".padStart(64, "0"),
+      },
+    ],
+  });
+
+  /*
+   * If there is a faucet, request (test) ETH if you have
+   * less than 1 ETH. Repeat every 20 seconds to ensure you don't
+   * run out.
+   */
+  if (networkConfig.faucetServiceUrl) {
+    const address = burnerAccount.address;
+    console.info("[Dev Faucet]: Player address -> ", address);
+
+    const faucet = createFaucetService(networkConfig.faucetServiceUrl);
+
+    const requestDrip = async () => {
+      const balance = await publicClient.getBalance({ address });
+      console.info(`[Dev Faucet]: Player balance -> ${balance}`);
+      const lowBalance = balance < parseEther("1");
+      if (lowBalance) {
+        console.info("[Dev Faucet]: Balance is low, dripping funds to player");
+        // Double drip
+        await faucet.dripDev({ address });
+        await faucet.dripDev({ address });
+      }
+    };
+
+    requestDrip();
+    // Request a drip every 20 seconds
+    setInterval(requestDrip, 20000);
+  }
+
+  return {
+    world,
+    components,
+    playerEntity: encodeEntity({ address: "address" }, { address: burnerWalletClient.account.address }),
+    publicClient,
+    walletClient: burnerWalletClient,
+    latestBlock$,
+    storedBlockLogs$,
+    waitForTransaction,
+    worldContract,
+    write$: write$.asObservable().pipe(share()),
+  };
+}
+```
+
+</CollapseCode>
+
+Click **Increment** a few times to see you only see the history for counter values 1 and 5.
+You can also go to the MUD Dev Tools and see that when you select **Components > :History** it only has those lines.
+
+### Explanation
+
+The `filters` field contains a list of filters.
+Only rows that match at least one line are synchronized.
+Each filter is a structure that can have up to three fields, and all the fields that are specified must match a row for the filter to match.
+
+- `tableId`, the table ID to synchronize.
+  You create this value using [`resourceToHex`](https://github.com/latticexyz/mud/blob/main/packages/common/src/resourceToHex.ts), the type can be either `table`, or `offchainTable`.
+- `key0`, the first key value (as a 32 byte hexadecimal string).
+- `key1`, the second key value (as a 32 byte hexadecimal string).
+
+<details>
+
+<summary>The filters in the code sample</summary>
+
+```ts
+    filters: [
+      {
+         tableId: resourceToHex({ type: "table", namespace: "", name: "Counter"}),
+      },
+```
+
+The first filter is for the `:Counter` table (`Counter` in the root namespace).
+We don't specify any keys, because we want all the rows of the table.
+It's a singleton so there is only one row anyway.
+
+```ts
+      {
+         tableId: resourceToHex({ type: "table", namespace: "", name: "History"}),
+         key0: "0x" + "1".padStart(64, "0")
+      },
+      {
+         tableId: resourceToHex({ type: "table", namespace: "", name: "History"}),
+         key0: "0x" + "5".padStart(64, "0")
```

Comment:
same here, let's use viem's `pad`, and we could also use viem's `toHex` (https://viem.sh/docs/utilities/toHex.html#tohex)